src/ast.rs

```diff
@@ -385,6 +403,14 @@ pub enum RawExpression {
 
     /// a binary expression, e.g. `s1:1 + s1:0`
     Binary(Box<Expression>, BinaryOperator, Box<Expression>),
+
+    /// 1 or more let expressions (i.e. the vector may be empty).
+    ///
+    /// Every binding is in scope in the bindings that come after it in the vector (i.e. a
+    /// binding is lexically scoped). The final expression, if present (and we expect it to
+    /// exist most of the time, otherwise, why bother with an if expression), can use all the
+    /// bindings. When absent, the value of the let expression is `()`.
+    Let(NonEmpty<LetBinding>, Option<Box<Expression>>),
```

Comment:
I vaguely remember offline discussion about this "chaining" of let exprs. If there were real tradeoffs considered, then consider putting some of those rationale here?

Reply:
See the comment in `parse_let_expression`:

```
    // We avoid a recursive parsing implementation of nested let expressions to avoid stack
    // overflows when fuzzing.
```

I don't think it makes sense to repeat the comment here. But if you insist, I will.

Reply:
No need to duplicate. I just missed it :)